tests/serializers/tests.py

```diff
@@ -410,6 +410,26 @@ def test_serialize_inherited_fields(self):
         self.assertEqual(self._get_field_values(child_data, "parent_m2m"), [])
         self.assertEqual(self._get_field_values(child_data, "parent_data"), [])
 
+    def test_serialize_only_pk(self):
+        with mock.patch("serializers.models.Article.categories") as mock_categories:
+            serializers.serialize(
+                self.serializer_name,
+                Article.objects.all(),
+                use_natural_foreign_keys=False,
+            )
+            for categories in mock_categories:
+                categories.only.assert_called_with("pk")
+
+    def test_serialize_no_only_with_natural_keys(self):
+        with mock.patch("serializers.models.Article.categories") as mock_categories:
+            serializers.serialize(
+                self.serializer_name,
+                Article.objects.all(),
+                use_natural_foreign_keys=True,
+            )
+            for categories in mock_categories:
+                categories.only.assert_not_called()
```

Comment:
I'd prefer not to use `mock` and check generated SQL instead. Also, `Category` doesn't have `natural_key()` so it works with `XML` serializer even when optimization is not use. Maybe:
```python
    def test_serialize_only_pk(self):
        with self.assertNumQueries(5) as ctx:
            serializers.serialize(
                self.serializer_name,
                Article.objects.all(),
                use_natural_foreign_keys=False,
            )
        categories_sql = ctx[1]["sql"]
        self.assertNotIn(connection.ops.quote_name("meta_data_id"), categories_sql)
        meta_data_sql = ctx[2]["sql"]
        self.assertNotIn(connection.ops.quote_name("kind"), meta_data_sql)

    def test_serialize_no_only_pk_with_natural_keys(self):
        with self.assertNumQueries(5) as ctx:
            serializers.serialize(
                self.serializer_name,
                Article.objects.all(),
                use_natural_foreign_keys=True,
            )
        categories_sql = ctx[1]["sql"]
        self.assertNotIn(connection.ops.quote_name("meta_data_id"), categories_sql)
        # CategoryMetaData with natural_key().
        meta_data_sql = ctx[2]["sql"]
        self.assertIn(connection.ops.quote_name("kind"), meta_data_sql)
```